### PR TITLE
Add an FAQ on torch.autograd.set_detect_anomaly(True)

### PR DESCRIPTION
**Pull request by obilaniu**

### CODE REVIEW

**Comment by obilaniu** [comment]
@satyaog I cannot explain it, but after adding the new text, the entire PyTorch FAQ disappeared. Any idea what's wrong with my rST syntax?

File: docs/Userguide_faq.rst [hunk 1]
```diff
@@ -157,3 +157,49 @@ be injected using the ``setup`` key in Hydra YAML config file::
...
+I am told my PyTorch job abuses the filesystem with extreme amounts of IOPS
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+A fairly common issue in PyTorch is::
+
+    RuntimeError: one of the variables needed for gradient computation has
+    been modified by an inplace operation: [torch.cuda.FloatTensor [1, 50, 300]],
+    which is output 0 of SplitBackward, is at version 2; expected version 0
+    instead. Hint: enable anomaly detection to find the operation that failed to
+    compute its gradient, with torch.autograd.set_detect_anomaly(True).
```
Comment (satyaog) [inline docs/Userguide_faq.rst:170]:
```suggestion
A fairly common issue in PyTorch is:

.. code:: none

   RuntimeError: one of the variables needed for gradient computation has been
   modified by an inplace operation: [torch.cuda.FloatTensor [1, 50, 300]],
   which is output 0 of SplitBackward, is at version 2; expected version 0
   instead. Hint: enable anomaly detection to find the operation that failed to
   compute its gradient, with torch.autograd.set_detect_anomaly(True).
```

File: docs/Userguide_faq.rst [hunk 2]
```diff
@@ -157,3 +157,49 @@ be injected using the ``setup`` key in Hydra YAML config file::
...
+
+PyTorch's autograd engine contains an "anomaly detection mode", which detects
+such things as NaN/infinities being created, and helps debugging in-place
+Tensor modifications. It is activated with
+
+.. code:: python
+
+    torch.autograd.set_detect_anomaly(True)
```
Comment (satyaog) [inline docs/Userguide_faq.rst:178]:
```suggestion
.. code:: python

   torch.autograd.set_detect_anomaly(True)
```

**Review by satyaog — APPROVED** [review]

**Comment by satyaog** [comment]
I can still the section here : https://mila-docs--162.org.readthedocs.build/en/162/Userguide.html#pytorch-issues

Looks good 👍 🎉

**Review by btravouillon — APPROVED** [review]
Looks good to me. 👍

**Pull request merged**